sdk/servicebus/service-bus/src/core/batchingReceiver.ts

```diff
@@ -395,6 +407,14 @@ export class BatchingReceiverLite {
         // at which point we'll wrap everything up and resolve the promise.
         receiver.drain = true;
         receiver.addCredit(1);
+
+        // WORKAROUND: currently addCredit(<positive number>) is the only way the next tick will actually
+        // send a flow frame. However, we don't want the extra credit we just added - it can result in an extra
+        // message being sent that we did not ask for (ie, it causes us to request remaining_credits + 1)
+        //
+        // This workaround goes in and "removes" the credit, leaving all the other necessary flags intact so a
+        // flow frame will still get sent. This is all just in-memory manipulation (nothing has been sent yet).
+        (receiver as RheaReceiverWithPrivateProperties)._link.credit--;
```

Comment:
Do we have an issue logged with rhea to allow calling `addCredit` with `0`? Anytime we reach into private internals of a dependency I hear the sound of a pager!

Reply:
I'll file one on Monday :)

Reply:
Getting the ball rolling:
https://github.com/amqp/rhea/issues/357

Reply:
PR merged: https://github.com/amqp/rhea/pull/358

Now just waiting for the rhea package release. 

After that, I'll update core-amqp (to pick up the new rhea package) and then update service bus to reference the new core-amqp.

Reply:
rhea was released (2.0.3!). I'll update the entire PR in just a moment to accomodate it.

Reply:
rhea-promise PR to expose the drain_credit() primitive: https://github.com/amqp/rhea-promise/pull/88

Reply:
Okay, update done!